drf_extra_fields/fields.py

```diff
@@ -103,7 +105,8 @@ class Base64ImageField(Base64FieldMixin, ImageField):
     INVALID_TYPE_MESSAGE = _("The type of the image couldn't be determined.")
 
     def get_file_extension(self, filename, decoded_file):
-        extension = imghdr.what(filename, decoded_file)
+        image = Image.open(io.BytesIO(decoded_file))
+        extension = image.format.lower()
```

Comment:
I think it's safer to use this as a fallback if imghdr fails to detect image extension. We know that imghdr working fine for most of the cases. Changing it with PIL might cause some other bugs.

- Could you please add comment about the bug and why we are using the PIL as fallback?
- Could you please add test for this case?

Reply:
Hi, Thank you for your kind feedback. I have added PIL as secondary option and kept imghdr primary. Also added a test case for it.